sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/models/TimeInterval.java

```diff
@@ -0,0 +1,264 @@
+    }
+
+    /**
+     * Creates an instance of {@link TimeInterval} using the start and end duration of the interval.
+     * @param startDuration The duration of the interval.
+     * @param endTime The end time of the interval.
+     */
+    public TimeInterval(Duration startDuration, OffsetDateTime endTime) {
+        this.endTime = Objects.requireNonNull(endTime, "'endTime' cannot be null");
+        this.startDuration = Objects.requireNonNull(startDuration, "'startDuration' cannot be null");
+        this.duration = null;
+        this.startTime = null;
+        this.endDuration = null;
+    }
+
+    /**
+     * Returns the duration of this {@link TimeInterval} instance.
+     * @return the duration of this {@link TimeInterval} instance.
+     */
+    public Duration getDuration() {
+        return duration;
+    }
+
+    /**
+     * Returns the start duration of this {@link TimeInterval} instance.
+     * @return the start duration of this {@link TimeInterval} instance.
+     */
+    public Duration getStartDuration() {
+        return startDuration;
+    }
+
+    /**
+     * Returns the end duration of this {@link TimeInterval} instance.
+     * @return the end duration of this {@link TimeInterval} instance.
+     */
+    public Duration getEndDuration() {
+        return endDuration;
+    }
+
+    /**
+     * Returns the start time of this {@link TimeInterval} instance.
+     * @return the start time of this {@link TimeInterval} instance.
+     */
+    public OffsetDateTime getStartTime() {
+        return startTime;
+    }
+
+    /**
+     * Returns the end time of this {@link TimeInterval} instance.
+     * @return the end time of this {@link TimeInterval} instance.
+     */
+    public OffsetDateTime getEndTime() {
+        return endTime;
+    }
+
+    /**
+     * Returns this {@link TimeInterval} in ISO 8601 string format.
+     * @return ISO 8601 formatted string representation of this {@link TimeInterval} instance.
+     */
+    public String toIso8601Format() {
+        if (startTime != null && endTime != null) {
+            return startTime + "/" + endTime;
+        }
+
+        if (startTime != null && endDuration != null) {
+            return startTime + "/" + endDuration;
+        }
+
+        if (startDuration != null && endTime != null) {
+            return startDuration + "/" + endTime;
+        }
+
+        if (duration != null) {
+            return duration.toString();
+        }
+        // should not reach here
+        throw LOGGER.logExceptionAsError(new IllegalStateException("Invalid date time range"));
+    }
+
+    /**
+     * This method takes an ISO 8601 formatted time interval string and returns an instance of {@link TimeInterval}.
+     * @param value The ISO 8601 formatted time interval string.
+     * @return An instance of {@link TimeInterval}.
+     * @throws IllegalArgumentException if {@code value} is not in the correct format.
+     */
+    public static TimeInterval parse(String value) {
```

Comment:
Is this type going to be used in serialization? If so, let's mark this as the JsonCreator

Reply:
I don't think this type is intended for serialization. Payloads should use the `toIso8601Format()` method to get an ISO standardized representation of time interval.